src/main/kotlin/ecommerce/services/ProductServiceCollection.kt

```diff
@@ -0,0 +1,75 @@
+package ecommerce.services
+
+import ecommerce.exception.NotFoundException
+import ecommerce.model.ProductDTO
+import ecommerce.model.ProductPatchDTO
+import org.springframework.stereotype.Service
+import java.util.concurrent.atomic.AtomicLong
+
+@Service
+class ProductServiceCollection : ProductService {
```

Comment:
By the way, how about annotating this class with `@Component` or `@Repository`?
Since it mainly deals with data rather than business logic, one of those might be more appropriate.